docs/recipes/atom-with-broadcast.mdx

```diff
@@ -13,17 +13,22 @@ By using the BroadcastChannel API, you can enable basic communication between br
 According to the MDN documentation, receiving a message during initialization is not supported in the BroadcastChannel, but if you want to support that functionality, you may need to add extra option to atomWithBroadcast, such as local storage.
 
 ```tsx
-import { atom } from 'jotai'
+import { atom, SetStateAction } from 'jotai'
 
 export function atomWithBroadcast<Value>(key: string, initialValue: Value) {
   const baseAtom = atom(initialValue)
   const listeners = new Set<(event: MessageEvent<any>) => void>()
   const channel = new BroadcastChannel(key)
+
   channel.onmessage = (event) => {
     listeners.forEach((l) => l(event))
   }
 
-  const broadcastAtom = atom<Value, { isEvent: boolean; value: Value }>(
+  const broadcastAtom = atom<
+    Value,
+    [{ isEvent: boolean; value: SetStateAction<Value> }],
+    void
+  >(
```

Comment:
Can we remove type here and add the type for `update` below? That's the recommended way.
```suggestion
  const broadcastAtom = atom(
```

Reply:
Sorry, what do you mean by `add the type for "update" below`? Do you mean doing this?

```ts
const broadcastAtom = atom(
  (get) => get(baseAtom),
  (get, set, update: { isEvent: boolean; value: SetStateAction<Value> }) => {
    set(baseAtom, update.value)

    if (!update.isEvent) {
      channel.postMessage(get(baseAtom))
    }
  }
)
```

Reply:
Yes, exactly! Sorry for the confusion.